pkg/sdk/go/consumers_test.go

```diff
@@ -65,65 +72,86 @@ func TestCreateSubscription(t *testing.T) {
-			token:        exampleUser1,
-			err:          nil,
+			subscription: sdkSubReq,
+			token:        validToken,
 			empty:        false,
-			id:           ID,
-			identifyErr:  nil,
-			userID:       validID,
+			svcReq:       notSubReq,
+			svcRes:       subID,
+			svcErr:       nil,
+			err:          nil,
 		},
 		{
 			desc:         "create new subscription with empty token",
-			subscription: sub1,
+			subscription: sdkSubReq,
 			token:        "",
-			err:          errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, apiutil.ErrBearerToken), http.StatusUnauthorized),
 			empty:        true,
-			id:           "",
-			identifyErr:  svcerr.ErrAuthorization,
+			svcReq:       notifiers.Subscription{},
+			svcRes:       "",
+			svcErr:       nil,
+			err:          errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, apiutil.ErrBearerToken), http.StatusUnauthorized),
 		},
 		{
 			desc:         "create new subscription with invalid token",
-			subscription: sub1,
-			token:        authmocks.InvalidValue,
-			err:          errors.NewSDKErrorWithStatus(svcerr.ErrAuthentication, http.StatusUnauthorized),
+			subscription: sdkSubReq,
+			token:        invalidToken,
 			empty:        true,
-			id:           "",
-			identifyErr:  svcerr.ErrAuthorization,
+			svcReq:       notSubReq,
+			svcRes:       "",
+			svcErr:       svcerr.ErrAuthentication,
+			err:          errors.NewSDKErrorWithStatus(svcerr.ErrAuthentication, http.StatusUnauthorized),
 		},
 		{
-			desc:         "create new empty subscription",
-			subscription: emptySubscription,
-			token:        token,
-			err:          errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, apiutil.ErrInvalidTopic), http.StatusBadRequest),
-			empty:        true,
-			id:           "",
-			identifyErr:  nil,
-			userID:       validID,
+			desc: "create new subscription with empty topic",
+			subscription: sdk.Subscription{
+				Topic:   "",
+				Contact: "contact",
+			},
+			token:  validToken,
+			empty:  true,
+			svcReq: notifiers.Subscription{},
+			svcErr: nil,
+			svcRes: "",
+			err:    errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, apiutil.ErrInvalidTopic), http.StatusBadRequest),
+		},
+		{
+			desc: "create new subscription with empty contact",
+			subscription: sdk.Subscription{
+				Topic:   "topic",
+				Contact: "",
+			},
+			token:  validToken,
+			empty:  true,
+			svcReq: notifiers.Subscription{},
+			svcErr: nil,
+			svcRes: "",
+			err:    errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, apiutil.ErrInvalidContact), http.StatusBadRequest),
 		},
 	}
 
 	for _, tc := range cases {
-		repoCall := auth.On("Identify", mock.Anything, &magistrala.IdentityReq{Token: tc.token}).Return(&magistrala.IdentityRes{Id: tc.userID}, tc.identifyErr)
-		repoCall1 := repo.On("Save", mock.Anything, mock.Anything).Return(tc.id, tc.err)
+		svcCall := nsvc.On("CreateSubscription", mock.Anything, tc.token, tc.svcReq).Return(tc.svcRes, tc.svcErr)
```

Comment:
Do the same https://github.com/absmach/magistrala/pull/2267#discussion_r1627472265